### ISSUE TITLE
[v0.27.0] Skaffold fix panic 

### ISSUE THREAD

**Issue by clmnt-wisepops** [GoogleContainerTools/skaffold#1964]
### Expected behavior
Skaffold fix should update skaffold.yaml file

### Actual behavior
Panic on run

### Information

- Skaffold version: v0.27.0
- Operating system: MacOS 10.14.4
- Contents of skaffold.yaml:

```yaml
apiVersion: skaffold/v1beta7
kind: Config
build:
  artifacts:
    - image: app
      context: .
      docker:
        dockerfile: ops/docker/app/Dockerfile
deploy:
  kubectl:
    manifests:
      - placeholder
      - ops/kubernetes/*.yaml
profiles:
  - name: local
    patches:
      - op: add
        path: /build/artifacts/0/docker
        value:
          buildArgs:
            PHP_INI: php.ini-development
      - op: add
        path: /build
        value:
          tagPolicy:
            sha256: {}
      - op: replace
        path: /deploy/kubectl/manifests/0
        value: ops/kubernetes/local/*.yaml
  - name: cluster
    patches:
      - op: add
        path: /build
        value:
          tagPolicy:
            gitCommit: {}
      - op: add
        path: /build
        value:
          googleCloudBuild:
            projectId: projectId
      - op: replace
        path: /deploy/kubectl/manifests/0
        value: ops/kubernetes/cluster/*.yaml
```

### Steps to reproduce the behavior

1. `skaffold fix`

```
WARN[0000] config version (skaffold/v1beta7) out of date: upgrading to latest (skaffold/v1beta8)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x14ebf83]

goroutine 1 [running]:
github.com/GoogleContainerTools/skaffold/vendor/gopkg.in/yaml%2ev2.handleErr(0xc0002bdc68)
        vendor/gopkg.in/yaml.v2/yaml.go:249 +0x9a
panic(0x218aa80, 0x34010c0)
        /usr/local/Cellar/go/1.12.4/libexec/src/runtime/panic.go:522 +0x1b5
github.com/GoogleContainerTools/skaffold/vendor/github.com/krishicks/yaml-patch.(*Node).MarshalYAML(0xc000595a20, 0x22273a0, 0xc000595a20, 0x3f82d50, 0xc000595a20)
        vendor/github.com/krishicks/yaml-patch/node.go:26 +0x33
github.com/GoogleContainerTools/skaffold/vendor/gopkg.in/yaml%2ev2.(*encoder).marshal(0xc0000de840, 0x0, 0x0, 0x22273a0, 0xc000572110, 0x196)
        vendor/gopkg.in/yaml.v2/encode.go:98 +0x82a
github.com/GoogleContainerTools/skaffold/vendor/gopkg.in/yaml%2ev2.(*encoder).structv.func1()
        vendor/gopkg.in/yaml.v2/encode.go:198 +0x55a
github.com/GoogleContainerTools/skaffold/vendor/gopkg.in/yaml%2ev2.(*encoder).mappingv(0xc0000de840, 0x0, 0x0, 0xc0002bcb80)
        vendor/gopkg.in/yaml.v2/encode.go:228 +0x147
github.com/GoogleContainerTools/skaffold/vendor/gopkg.in/yaml%2ev2.(*encoder).structv(0xc0000de840, 0x0, 0x0, 0x2267da0, 0xc0005720e0, 0x199)
        vendor/gopkg.in/yaml.v2/encode.go:185 +0xf7
github.com/GoogleContainerTools/skaffold/vendor/gopkg.in/yaml%2ev2.(*encoder).marshal(0xc0000de840, 0x0, 0x0, 0x2267da0, 0xc0005720e0, 0x199)
        vendor/gopkg.in/yaml.v2/encode.go:132 +0x726
github.com/GoogleContainerTools/skaffold/vendor/gopkg.in/yaml%2ev2.(*encoder).slicev(0xc0000de840, 0x0, 0x0, 0x20c2ac0, 0xc0000b4b30, 0x197)
        vendor/gopkg.in/yaml.v2/encode.go:244 +0x1f8
github.com/GoogleContainerTools/skaffold/vendor/gopkg.in/yaml%2ev2.(*encoder).marshal(0xc0000de840, 0x0, 0x0, 0x20c2ac0, 0xc0000b4b30, 0x197)
        vendor/gopkg.in/yaml.v2/encode.go:138 +0x412
github.com/GoogleContainerTools/skaffold/vendor/gopkg.in/yaml%2ev2.(*encoder).structv.func1()
        vendor/gopkg.in/yaml.v2/encode.go:198 +0x55a
github.com/GoogleContainerTools/skaffold/vendor/gopkg.in/yaml%2ev2.(*encoder).mappingv(0xc0000de840, 0x0, 0x0, 0xc0002bd108)
        vendor/gopkg.in/yaml.v2/encode.go:228 +0x147
github.com/GoogleContainerTools/skaffold/vendor/gopkg.in/yaml%2ev2.(*encoder).structv(0xc0000de840, 0x0, 0x0, 0x2267e60, 0xc0000b4a80, 0x199)
        vendor/gopkg.in/yaml.v2/encode.go:185 +0xf7
github.com/GoogleContainerTools/skaffold/vendor/gopkg.in/yaml%2ev2.(*encoder).marshal(0xc0000de840, 0x0, 0x0, 0x2267e60, 0xc0000b4a80, 0x199)
        vendor/gopkg.in/yaml.v2/encode.go:132 +0x726
github.com/GoogleContainerTools/skaffold/vendor/gopkg.in/yaml%2ev2.(*encoder).slicev(0xc0000de840, 0x0, 0x0, 0x20c2b00, 0xc000572360, 0x197)
        vendor/gopkg.in/yaml.v2/encode.go:244 +0x1f8
github.com/GoogleContainerTools/skaffold/vendor/gopkg.in/yaml%2ev2.(*encoder).marshal(0xc0000de840, 0x0, 0x0, 0x20c2b00, 0xc000572360, 0x197)
        vendor/gopkg.in/yaml.v2/encode.go:138 +0x412
github.com/GoogleContainerTools/skaffold/vendor/gopkg.in/yaml%2ev2.(*encoder).structv.func1()
        vendor/gopkg.in/yaml.v2/encode.go:198 +0x55a
github.com/GoogleContainerTools/skaffold/vendor/gopkg.in/yaml%2ev2.(*encoder).mappingv(0xc0000de840, 0x0, 0x0, 0xc0002bd690)
        vendor/gopkg.in/yaml.v2/encode.go:228 +0x147
github.com/GoogleContainerTools/skaffold/vendor/gopkg.in/yaml%2ev2.(*encoder).structv(0xc0000de840, 0x0, 0x0, 0x2267f20, 0xc0005722a0, 0x199)
        vendor/gopkg.in/yaml.v2/encode.go:185 +0xf7
github.com/GoogleContainerTools/skaffold/vendor/gopkg.in/yaml%2ev2.(*encoder).marshal(0xc0000de840, 0x0, 0x0, 0x2267f20, 0xc0005722a0, 0x199)
        vendor/gopkg.in/yaml.v2/encode.go:132 +0x726
github.com/GoogleContainerTools/skaffold/vendor/gopkg.in/yaml%2ev2.(*encoder).marshal(0xc0000de840, 0x0, 0x0, 0x21921e0, 0xc0005722a0, 0x16)
        vendor/gopkg.in/yaml.v2/encode.go:126 +0x59f
github.com/GoogleContainerTools/skaffold/vendor/gopkg.in/yaml%2ev2.(*encoder).marshalDoc(0xc0000de840, 0x0, 0x0, 0x21921e0, 0xc0005722a0, 0x16)
        vendor/gopkg.in/yaml.v2/encode.go:80 +0x10f
github.com/GoogleContainerTools/skaffold/vendor/gopkg.in/yaml%2ev2.Marshal(0x21921e0, 0xc0005722a0, 0x0, 0x0, 0x0, 0x0, 0x0)
        vendor/gopkg.in/yaml.v2/yaml.go:203 +0x329
github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/cmd.runFix(0x25b7220, 0xc0000bc000, 0x23878f8, 0xd, 0xc00050fc00, 0x0, 0x0)
        cmd/skaffold/app/cmd/fix.go:61 +0x10e
github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/cmd.NewCmdFix.func1(0xc000204500, 0x3446000, 0x0, 0x0, 0x0, 0x0)
        cmd/skaffold/app/cmd/fix.go:36 +0x56
github.com/GoogleContainerTools/skaffold/vendor/github.com/spf13/cobra.(*Command).execute(0xc000204500, 0x3446000, 0x0, 0x0, 0xc000204500, 0x3446000)
        vendor/github.com/spf13/cobra/command.go:762 +0x465
github.com/GoogleContainerTools/skaffold/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc00019cc80, 0xc0000bc000, 0x25b7220, 0xc0000c0000)
        vendor/github.com/spf13/cobra/command.go:852 +0x2ec
github.com/GoogleContainerTools/skaffold/vendor/github.com/spf13/cobra.(*Command).Execute(...)
        vendor/github.com/spf13/cobra/command.go:800
github.com/GoogleContainerTools/skaffold/cmd/skaffold/app.Run(0x25b7220, 0xc0000bc000, 0x25b7220, 0xc0000c0000, 0xc00050ff88, 0x100554f)
        cmd/skaffold/app/skaffold.go:35 +0xd3
main.main()
        cmd/skaffold/skaffold.go:30 +0x4e
```

**Comment by MisterTimn** [comment]
+1 encountering the same issue after updating skaffold and trying to fix my .yaml

**Comment by corneliusweig** [comment]
That one goes on me :(
Looks like the `Node` struct has some unexported fields which are accessed via reflection.

**Comment by corneliusweig** [comment]
Seems I was not the only culprit :relieved:
Fixing the unexported field access is straightforward (see #1971). However, the whole schema upgrade is broken, because it relies on json serialization and `vendor/github.com/krishicks/yaml-patch/node.go:7` does not serialize well, because it results in a `Node` with `raw=nil` content. This is later dereferenced, causing a segfault. So upgrades with profile patches were always broken.

**Comment by corneliusweig** [comment]
> That one goes on me :(
> Looks like the `Node` struct has some unexported fields which are accessed via reflection.

I was confused.. This is another problem in the current master. The problem above has been much older, but will be fixed with #1971.
